tests/apollo/util/test_base.py

```diff
@@ -51,22 +51,31 @@ async def wrapper(*args, **kwargs):
 
     return decorator
 
-def retry_test(max_retries: int):
-    """
-    Runs a test multiple times until a run succeeds
+def repeat_test(max_repeats: int, break_on_first_failure: bool, break_on_first_success: bool):
     """
+    Runs a test  max_repeats times when both break_on_first_failure and break_on_first_success et to False.
+    Only one of break_on_first_failure or break_on_first_success can be True (both can be False).
 
+    break_on_first_success is True: run test up to max_repeats times and breaks after the 1st successful test.
+    break_on_first_failure is True: run test up to max_repeats times and breaks after the 1st failing test.
+    """
+    assert not (break_on_first_failure and break_on_first_success), \
+        "both flags break_on_first_failure and break_on_first_success cannot be enabled at the same time!"
     def decorator(async_fn):
         @wraps(async_fn)
         async def wrapper(*args, **kwargs):
-            for i in range(1, max_retries + 1):
+            for i in range(1, max_repeats + 1):
                 try:
+                    if (max_repeats > 1):
+                        print(f"Running iteration {i}/{max_repeats}, (break_on_first_failure={break_on_first_failure},"
```

Comment:
Idk why, but I can't comment on the previous discussion.
If the reason for printing is that the logs are not easily readable, this is no longer the case if you use the APOLLO_LOG_STDOUT environment variable.
I'm generally against mixing different types of reports as it means the searching for messages becomes much harder.